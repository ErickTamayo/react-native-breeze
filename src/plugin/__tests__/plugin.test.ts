import { transform, TransformOptions } from "babel-core";

const plugin = require("../");
const opts: TransformOptions = {
  presets: ["react-native"],
  plugins: [plugin],
  compact: false,
};

describe("react-native-breeze plugin", () => {
  describe("br", () => {
    it("should compile regular style", () => {
      const input = `
        import { View } from "react-native";

        const Component = () => {
          return <View style={{ backgroundColor: "red" }} />;
        };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should compile breeze styles", () => {
      const input = `
        import { View } from "react-native";
        import { br } from "react-native-breeze";
      
        const Calendar = () => {
          return <View style={br\`bg-red-500\`} />;
        };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should compile both breeze and regular styles", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = () => {
      return <View style={[br\`bg-red-500\`, { height: 10 }]} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should Error if br declaration is outside of a Component", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const style = br\`Ã¥bg-red-500\`

    const Calendar = () => {
      return <View style={{}} />;
    };`;

      expect(() => transform(input, opts)).toThrow();
    });

    it("should apply hover correctly", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = () => {
      return <View style={br\`hover:bg-red-500\`} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply hover if there is a prop already correctly", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = () => {

      const handleOnMouseEnter = () => {};
      const handleOnMouseLeave = () => {};

      return <View style={br\`hover:bg-red-500\`} onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply hover if there is an spread and a prop already correctly", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = (props) => {

      const handleOnMouseEnter = () => {};
      const handleOnMouseLeave = () => {};

      return <View style={br\`hover:bg-red-500\`} onMouseEnter={handleOnMouseEnter} onMouseLeave={handleOnMouseLeave} {...props} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply hover handlers to spread proactively", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = (props) => {
      return <View style={br\`hover:bg-red-500\`} {...props} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply hover handlers to mutiple spread proactively", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = (props) => {

      const otherProps = {};

      return <View style={br\`hover:bg-red-500\`} {...props} {...otherProps}/>;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should not apply hover if var is not binded to a JSX", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";

    const Calendar = () => {
      const noBind = br\`hover:bg-red-500\`
      return <View style={{}} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it.todo("should apply hover if a var is binded to a JSX");

    it("should apply focus correctly", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = () => {
      return <View style={br\`focus:bg-red-500\`} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply focus if there is a prop already correctly", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = () => {

      const handleOnFocus = () => {};
      const handleOnBlur = () => {};

      return <View style={br\`focus:bg-red-500\`} onFocus={handleOnFocus} onBlur={handleOnBlur} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply focus if there is an spread and a prop already correctly", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = (props) => {

      const handleOnFocus = () => {};
      const handleOnBlur = () => {};

      return <View style={br\`focus:bg-red-500\`} onFocus={handleOnFocus} onBlur={handleOnBlur} {...props} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply focus handlers to spread proactively", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = (props) => {
      return <View style={br\`focus:bg-red-500\`} {...props} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should apply focus handlers to mutiple spread proactively", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";
  
    const Calendar = (props) => {

      const otherProps = {};

      return <View style={br\`focus:bg-red-500\`} {...props} {...otherProps}/>;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it("should not apply focus if var is not binded to a JSX", () => {
      const input = `
    import { View } from "react-native";
    import { br } from "react-native-breeze";

    const Calendar = () => {
      const noBind = br\`focus:bg-red-500\`
      return <View style={{}} />;
    };`;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });

    it.todo("should apply focus if a var is binded to a JSX");
  });

  describe("br.value", () => {
    it("should compile value correctly", () => {
      const input = `
      import { br } from "react-native-breeze";
      const value = br.value\`bg-red-500\`
    `;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });
  });

  describe("br.raw", () => {
    it("should compile value correctly", () => {
      const input = `
      import { br } from "react-native-breeze";
      const raw = br.raw\`bg-red-500\`
    `;

      const { code } = transform(input, opts);
      expect(code).toMatchSnapshot();
    });
  });
});
