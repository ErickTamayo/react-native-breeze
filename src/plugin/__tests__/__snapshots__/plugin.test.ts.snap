// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-native-breeze plugin br should apply focus correctly 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{focus:{backgroundColor:\\"#f56565\\"}}}});

  return React.createElement(_reactNative.View, { style: _hook.style, onFocus: _hook.handleOnFocus(),
    onBlur: _hook.handleOnBlur()
  });
};"
`;

exports[`react-native-breeze plugin br should apply focus handlers to mutiple spread proactively 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar(props) {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{focus:{backgroundColor:\\"#f56565\\"}}}});

  var otherProps = {};

  return React.createElement(_reactNative.View, _extends({ style: _hook.style }, props, otherProps, {
    onFocus: _hook.handleOnFocus(otherProps.onFocus || props.onFocus),
    onBlur: _hook.handleOnBlur(otherProps.onBlur || props.onBlur)
  }));
};"
`;

exports[`react-native-breeze plugin br should apply focus handlers to spread proactively 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar(props) {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{focus:{backgroundColor:\\"#f56565\\"}}}});

  return React.createElement(_reactNative.View, _extends({ style: _hook.style }, props, {
    onFocus: _hook.handleOnFocus(props.onFocus),
    onBlur: _hook.handleOnBlur(props.onBlur)
  }));
};"
`;

exports[`react-native-breeze plugin br should apply focus if there is a prop already correctly 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{focus:{backgroundColor:\\"#f56565\\"}}}});

  var handleOnFocus = function handleOnFocus() {};
  var handleOnBlur = function handleOnBlur() {};

  return React.createElement(_reactNative.View, { style: _hook.style, onFocus: _hook.handleOnFocus(handleOnFocus),
    onBlur: _hook.handleOnBlur(handleOnBlur)
  });
};"
`;

exports[`react-native-breeze plugin br should apply focus if there is an spread and a prop already correctly 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar(props) {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{focus:{backgroundColor:\\"#f56565\\"}}}});

  var handleOnFocus = function handleOnFocus() {};
  var handleOnBlur = function handleOnBlur() {};

  return React.createElement(_reactNative.View, _extends({ style: _hook.style }, props, {
    onFocus: _hook.handleOnFocus(handleOnFocus),
    onBlur: _hook.handleOnBlur(handleOnBlur)
  }));
};"
`;

exports[`react-native-breeze plugin br should apply hover correctly 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{hover:{backgroundColor:\\"#f56565\\"}}}});

  return React.createElement(_reactNative.View, { style: _hook.style, onMouseEnter: _hook.handleOnMouseEnter(),
    onMouseLeave: _hook.handleOnMouseLeave()
  });
};"
`;

exports[`react-native-breeze plugin br should apply hover handlers to mutiple spread proactively 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar(props) {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{hover:{backgroundColor:\\"#f56565\\"}}}});

  var otherProps = {};

  return React.createElement(_reactNative.View, _extends({ style: _hook.style }, props, otherProps, {
    onMouseEnter: _hook.handleOnMouseEnter(otherProps.onMouseEnter || props.onMouseEnter),
    onMouseLeave: _hook.handleOnMouseLeave(otherProps.onMouseLeave || props.onMouseLeave)
  }));
};"
`;

exports[`react-native-breeze plugin br should apply hover handlers to spread proactively 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar(props) {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{hover:{backgroundColor:\\"#f56565\\"}}}});

  return React.createElement(_reactNative.View, _extends({ style: _hook.style }, props, {
    onMouseEnter: _hook.handleOnMouseEnter(props.onMouseEnter),
    onMouseLeave: _hook.handleOnMouseLeave(props.onMouseLeave)
  }));
};"
`;

exports[`react-native-breeze plugin br should apply hover if there is a prop already correctly 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{hover:{backgroundColor:\\"#f56565\\"}}}});

  var handleOnMouseEnter = function handleOnMouseEnter() {};
  var handleOnMouseLeave = function handleOnMouseLeave() {};

  return React.createElement(_reactNative.View, { style: _hook.style, onMouseEnter: _hook.handleOnMouseEnter(handleOnMouseEnter),
    onMouseLeave: _hook.handleOnMouseLeave(handleOnMouseLeave)
  });
};"
`;

exports[`react-native-breeze plugin br should apply hover if there is an spread and a prop already correctly 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar(props) {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{hover:{backgroundColor:\\"#f56565\\"}}}});

  var handleOnMouseEnter = function handleOnMouseEnter() {};
  var handleOnMouseLeave = function handleOnMouseLeave() {};

  return React.createElement(_reactNative.View, _extends({ style: _hook.style }, props, {
    onMouseEnter: _hook.handleOnMouseEnter(handleOnMouseEnter),
    onMouseLeave: _hook.handleOnMouseLeave(handleOnMouseLeave)
  }));
};"
`;

exports[`react-native-breeze plugin br should compile both breeze and regular styles 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{base:{backgroundColor:\\"#f56565\\"}}}});

  return React.createElement(_reactNative.View, { style: [_hook.style, { height: 10 }] });
};"
`;

exports[`react-native-breeze plugin br should compile breeze styles 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{base:{backgroundColor:\\"#f56565\\"}}}});

  return React.createElement(_reactNative.View, { style: _hook.style });
};"
`;

exports[`react-native-breeze plugin br should compile regular style 1`] = `
"var _reactNative = require(\\"react-native\\");

var Component = function Component() {
  return React.createElement(_reactNative.View, { style: { backgroundColor: \\"red\\" } });
};"
`;

exports[`react-native-breeze plugin br should not apply focus if var is not binded to a JSX 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{focus:{backgroundColor:\\"#f56565\\"}}}});

  var noBind = _hook.style;
  return React.createElement(_reactNative.View, { style: {} });
};"
`;

exports[`react-native-breeze plugin br should not apply hover if var is not binded to a JSX 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var _reactNative = require(\\"react-native\\");

var Calendar = function Calendar() {
  var _hook = (0, _reactNativeBreeze.useBreeze)({default:{all:{hover:{backgroundColor:\\"#f56565\\"}}}});

  var noBind = _hook.style;
  return React.createElement(_reactNative.View, { style: {} });
};"
`;

exports[`react-native-breeze plugin br.raw should compile value correctly 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var raw = {
      backgroundColor: \\"#f56565\\"
};"
`;

exports[`react-native-breeze plugin br.value should compile value correctly 1`] = `
"var _reactNativeBreeze = require(\\"react-native-breeze\\");

var value = \\"#f56565\\";"
`;
